src/styles/main.scss

```diff
@@ -1,3 +1,62 @@
+@import "variables";
+
 body {
   margin: 0;
 }
+
+* {
+  box-sizing: border-box;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $width-cell * 7 + $padding-cal * 2 + $gap * 6;
+  margin: 20px auto 0;
+  gap: $gap;
+  padding: $padding-cal ;
+  box-sizing: border-box;
+
+  @each $day , $num in $days {
+    &--start-#{$day} :first-child {
+      margin-left: ($width-cell + $gap) * $num;
+    }
+  }
+
+  @for $day from 28 to 31 {
+    &--month-#{$day} &__day:nth-child(n + #{$day + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    background-color: #eee;
+    width: $width-cell ;
+    height: $height-cell;
+    border: 1px solid grey;
+    box-sizing: border-box;
+    position: relative;
+    transition: all 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i}):before {
+        color: #000;
+        font-size: 30px;
+        font-family: Arial, Helvetica, sans-serif;
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+        display: block;
```

Comment:
- You don't need to specify these properties to each `:nth-child(#{$i})`. Just once to `:before` selector
- instead of `position: absolute;` and other unnecessary manipulations use flex to `&__day` selector